src/time/time.h

```diff
@@ -1,95 +1,114 @@
 #pragma once
 
-/* TODO(yao): move this into ccommon:
- *   It is common to have a TTL/age for keys in key-value store. A wrapper
- *   like this often achieves two goals: 1) it provides a process-local, cached
- *   time value so we don't need to call the relatively expensive time syscalls
- *   too often; 2) since we are already using a local timer, the zero point of
- *   the timer can be set for the process to simplify operations like timestamp
- *   comparison, expiration, etc.
- *
- *   The existing caching solutions have timestamps of vaiours granularity and
- *   definition: e.g. Redis has high resolution TTL, while memcached time is at
- *   second-level granularity. To remain protocol compatible with these imple-
- *   mentations, we may need more than one time wrapper. And even more may be
- *   added in the future to strike different balance between precision and cost.
- */
-#include <cc_define.h>
-
-#include <inttypes.h>
+#include <stdint.h>
 #include <time.h>
 
-/* NOTE(yao): this whole time module needs a major overhaul */
+#include "util/define.h"
 
 /*
- * How time works internally for memcached
- *
- * Memcached server uses a timer with second granularity, which starts as the
- * process starts, and is set to 2 initially to avoid having 0 aged items.
- *
- * On systems where size(time_t) > sizeof(uint32_t), this gives
- * us space savings over tracking absolute unix time of type time_t
+ * For less granular time, rel_time_t will suffice and gives space savings.
+ * For more granular time, rel_time_fine_t gives additional precision.
  */
 typedef uint32_t rel_time_t;
```

Comment:
I think the original typedef wasn't necessarily the best choice of name. Is this relative because the process itself is started at arbitrary time measured by Unix time, or is this relative because we are comparing against the process's clock? I would like to use this as an opportunity to improve that naming, e.g., how about `typedef uint32_t process_time_u` (also fixing the naming convention according to our style guide to move away from `_t` in applications). So now we know we are talking about time whose zero point is set to the time the process started.